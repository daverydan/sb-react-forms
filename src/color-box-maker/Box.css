.Box {
  margin: 30px auto;
  position: relative;
}

.Box button {
  background: #fff;
  border-radius: 50%;
  border: 1px solid #000;
  height: 24px;
  line-height: 20px;
  position: absolute;
  right: -12px;
  top: -12px;
  width: 24px;
}

.Box button:hover {
  background: #e7e7e7;
  cursor: pointer;
}
