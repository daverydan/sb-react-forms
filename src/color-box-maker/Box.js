import React from "react";
import "./Box.css";

const Box = ({
  id = 0,
  bgColor = "green",
  height = 50,
  width = 50,
  removeBox,
}) => {
  return (
    <div
      id={`box-${id}`}
      className="Box"
      style={{ backgroundColor: bgColor, height: height, width: width }}
    >
      <button onClick={removeBox} data-testid={`box-btn-${id}`}>
        x
      </button>
    </div>
  );
};

export default Box;
