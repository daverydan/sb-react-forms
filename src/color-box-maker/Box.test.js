import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

test("app not broken", () => {
  render(<Box />);
});

test("it matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
