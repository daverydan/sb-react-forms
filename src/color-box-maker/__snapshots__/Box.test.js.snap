// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="background-color: green; height: 50px; width: 50px;"
  >
    <button>
      x
    </button>
  </div>
</DocumentFragment>
`;
